library/index.js

```diff
@@ -0,0 +1,35 @@
+const   { WebClient }           = require('@slack/web-api'),
+        { createEventAdapter }  = require('@slack/events-api'),
+        { createReadStream }    = require('fs');
+
+
+// get access token from slack bot and set you enviroment variable SLACK_TOKEN using export SLACK_TOKEN="slack token" in your terminal
+const token = process.env.SLACK_TOKEN;
+const slackEvents = createEventAdapter(process.env.SLACK_SIGNING_SECRET);
+const web = new WebClient(token);
+
+// This argument can be a channel ID, a DM ID, a MPDM ID, or a group ID.
+const channelId = 'C01AA5U5BGT';
+
+
+// sends file to slack works for both private and group depends on the channel id
+async function sendFile(filename,channelId){
+    const response = await web.files.upload({
+        filename, file: createReadStream(filename), channels: channelId
+    })
+    //we can collect from data from the response such as slack link to file, id, etc
+    //console.log(response);
+}
+
+// sets up server if you havent set a port it defaults to 3000
+const port = process.env.PORT || 3000;
+
+
+// starts server and waits for events to arrive
+(async () => {
+    const server = await slackEvents.start(port);
+    console.log("Server starter on port %s", port);
+  })();
```

Comment:
If this package is supposed to be a lib, maybe we could move the server start code away from the lib code. So instead we have an an anonymous function here, we maybe could have a `startSlackBotListener` function and our backend service call this function to start the event listener. Just a thought, what do you think how we should organize the code to separate the library and web service ?

Reply:
Yeah that's what I was looking into. Since I started having issues with the retrieving data back from form submissions because this was running and Events API seems to not be able to parse interactive forms.